### PR TITLE
Added system tests for directions

### PR DESCRIPTION
**Pull request by pobudzey**

### CODE REVIEW

File: app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt
Comment (felixlapierre) [inline app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt]: I think this one was used in several test classes now, should we extract it into a SystemTestUtils class?

File: app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt
Comment (felixlapierre) [inline app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt]: There is a bit of debate surrounding randomness in tests. Personally, i'd rather we remove this kind of randomness and always use one particular travel mode.

The reasoning behind this is that if the feature is broken for, say, only "driving" mode, then the test will be flaky and fail 33% of the time. When it does fail intermittently, we'll have no idea why until we dig in and find this element of randomness, which could waste a lot of time. It could also lie unnoticed through a merge into Master, and the dev who notices it may not be the dev who broke it.

All this to say: just put travelMode = "walking" or something

File: app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt
Comment (felixlapierre) [inline app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt:100]: Any way to assert that the start and end are at the correct latitude/longitude? Or is that too difficult?

File: app/src/androidTest/java/com/example/campusguide/SwitchCampusSystemTest.kt
Comment (felixlapierre) [inline app/src/androidTest/java/com/example/campusguide/SwitchCampusSystemTest.kt]: Given how many places had to be refactored to accomodate changing this destriptor, i think we need to move it into a constant or something. I know it changes based on which map it is, but we still need to make it easier to adjust without breaking stuff.

**Review by felixlapierre — APPROVED** [review]
Well done!!

I have a few little concerns but nothing major, so I'll approve this and leave it to you to address them before you merge 😄

File: app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt
Comment (pobudzey) [reply app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt]: Good point, done

File: app/src/androidTest/java/com/example/campusguide/SwitchCampusSystemTest.kt
Comment (pobudzey) [reply app/src/androidTest/java/com/example/campusguide/SwitchCampusSystemTest.kt]: Done

File: app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt
Comment (pobudzey) [reply app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt]: I agree with most of what you said
However, functionally all three travel mode buttons do the same thing  (they make an API request and display the route) so if one of them works, then most likely all of them will (and vice versa) which is why I decided to go with the randomness approach and avoid writing duplicated code to test every travel mode

File: app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt
Comment (pobudzey) [reply app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt:100]: Not worth the effort imo

**Pull request merged**
